wifiphisher/accesspoint.py

```diff
@@ -0,0 +1,489 @@
+"""
+This module handles all the operations regarding locating all the
+available access points
+"""
+
+from __future__ import division
+import threading
+import time
+import logging
+import scapy.layers.dot11 as dot11
+import constants
+
+MODULE_LOGGER = logging.getLogger("wifiphisher.accesspoint")
+
+
+class AccessPoint(object):
+    """ This class represents an access point """
+
+    def __init__(self, ssid, bssid, channel, encryption):
+        """
+        Setup the class with all the given arguments
+
+        :param self: An AccessPoint object
+        :param ssid: The name of the access point
+        :param bssid: The MAC address of the access point
+        :param channel: The channel number of the access point
+        :param encryption: The encryption type of the access point
+        :type self: AccessPoint
+        :type ssid: string
+        :type bssid: string
+        :type channel: string
+        :type encryption: string
+        """
+
+        self._name = ssid
+        self._mac_address = bssid
+        self._channel = channel
+        self._encryption = encryption
+        self._signal_strength = None
+        self._clients = set()
+        self.logger = logging.getLogger("wifiphisher.accesspoint.AccessPoint")
+
+        self.logger.info("Creating AccessPoint object")
+
+    def get_name(self):
+        """
+        Return the name(ESSID) of the access point
+
+        :param self: An AccessPoint object
+        :type self: AccessPoint
+        :return: Name of the access point
+        :rtype: string
+        """
+
+        self.logger.info("Returning name")
```

Comment:
perhaps we can leave as is for now and once we design the logging we can modify it.